svf/lib/Util/SVFBugReport.cpp

```diff
@@ -0,0 +1,326 @@
+}
+
+void DoubleFreeBug::printBugToTerminal()
+{
+    SVFUtil::errs() << SVFUtil::bugMsg2("\t Double Free :") <<  " memory allocation at : ("
+                    << bugInst->getSourceLoc() << ")\n";
+
+    SVFUtil::errs() << "\t\t double free path: \n";
+    for(Set<std::string>::iterator iter = doubleFreePath.begin(), eiter = doubleFreePath.end();
+         iter!=eiter; ++iter)
+    {
+        SVFUtil::errs() << "\t\t  --> (" << *iter << ") \n";
+    }
+    SVFUtil::errs() << "\n";
+}
+
+cJSON * FileNeverCloseBug::getBugDescription()
+{
+    cJSON *bugDescription = cJSON_CreateObject();
+    return bugDescription;
+}
+
+void FileNeverCloseBug::printBugToTerminal()
+{
+    SVFUtil::errs() << SVFUtil::bugMsg1("\t FileNeverClose :") <<  " file open location at : ("
+                    << bugInst->getSourceLoc() << ")\n";
+}
+
+cJSON * FilePartialCloseBug::getBugDescription()
+{
+    cJSON *bugDescription = cJSON_CreateObject();
+
+    cJSON *pathInfo = cJSON_CreateArray();
+    auto pathIt = conditionalFileClosePath.begin();
+    for(; pathIt != conditionalFileClosePath.end(); pathIt++){
+        cJSON *newPath = cJSON_CreateString((*pathIt).c_str());
+        cJSON_AddItemToArray(pathInfo, newPath);
+    }
+    cJSON_AddItemToObject(bugDescription, "ConditionalFileClosePath", pathInfo);
+
+    return bugDescription;
+}
+
+void FilePartialCloseBug::printBugToTerminal()
+{
+    SVFUtil::errs() << SVFUtil::bugMsg2("\t PartialFileClose :") <<  " file open location at : ("
+                    << bugInst->getSourceLoc() << ")\n";
+
+    SVFUtil::errs() << "\t\t conditional file close path: \n";
+    for(Set<std::string>::iterator iter = conditionalFileClosePath.begin(), eiter = conditionalFileClosePath.end();
+         iter!=eiter; ++iter)
+    {
+        SVFUtil::errs() << "\t\t  --> (" << *iter << ") \n";
+    }
+    SVFUtil::errs() << "\n";
+}
+
+const std::string CallSiteEvent::getFuncName() const
+{
+    return callSite->getCallSite()->getFunction()->getName();
+}
+
+const std::string CallSiteEvent::getEventLoc() const
+{
+    return callSite->getCallSite()->getSourceLoc();
+}
+
+const std::string CallSiteEvent::getEventDescription() const
+{
+    std::string description("calls ");
+    const SVFFunction *callee = SVFUtil::getCallee(callSite->getCallSite());
+    if(callee == nullptr){
+        description += "<unknown>";
+    }else{
+        description += callee->getName();
+    }
+    return description;
+}
+
+const std::string BranchEvent::getFuncName() const
+{
+    return branchStmt->getInst()->getFunction()->getName();
+}
+
+const std::string BranchEvent::getEventLoc() const
+{
+    return branchStmt->getInst()->getSourceLoc();
+}
+
+const std::string BranchEvent::getEventDescription() const
+{
+    return description;
+}
+
+SVFBugReport::~SVFBugReport()
+{
+    auto bugIt = bugVector.begin();
+    for(;bugIt != bugVector.end(); bugIt ++){
+        delete (*bugIt);
+    }
+}
+
+std::string SVFBugReport::toString()
```

Comment:
better change the method to "void SVFBugReport::toJson()". Never return large string!!!